include/triton/core/tritonrepoagent.h

```diff
@@ -0,0 +1,307 @@
+///     being loaded.
+///
+///   TRITONREPOAGENT_ACTION_RELOAD: A currently loaded model is
+///     being reloaded.
+///
+typedef enum TRITONREPOAGENT_actiontype_enum {
+  TRITONREPOAGENT_ACTION_LOAD,
+  TRITONREPOAGENT_ACTION_RELOAD
+} TRITONREPOAGENT_ActionType;
+
+/// Get the location of the files that make up the model. The
+/// 'location' communicated depends on how the model is being
+/// communicated to the agent as indicated by 'artifact_type'.
+///
+///   TRITONREPOAGENT_ARTIFACT_FILESYSTEM: The model artifacts are
+///     made available to the agent via the local
+///     filesytem. 'location' returns the full path to the directory
+///     in the model repository that contains the model's
+///     artifacts. The returned location string is owned by Triton,
+///     not the caller, and so should not be modified or freed. The
+///     contents of the directory are owned by Triton, not the agent,
+///     and so the agent should not delete or modify the contents. Use
+///     TRITONREPOAGENT_RepositoryAcquire and
+///     TRITONREPOAGENT_RepositoryCommit to get a location that can be
+///     used to modify the model repository contents.
+///
+/// \param agent The agent.
+/// \param model The model.
+/// \param artifact_type Returns the artifact type for the location.
+/// \param path Returns the location.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONREPOAGENT_DECLSPEC TRITONSERVER_Error*
+TRITONREPOAGENT_ModelRepositoryLocation(
+    TRITONREPOAGENT_Agent* agent, TRITONREPOAGENT_AgentModel* model,
+    TRITONREPOAGENT_ArtifactType* artifact_type, const char** location);
+
+/// Acquire a location where the agent can produce a new version of
+/// the model repository files. The agent takes ownership of the
+/// location until it is released by calling
+/// TRITONREPOAGENT_ModelRepositoryLocationCommit or
+/// TRITONREPOAGENT_ModelRepositoryLocationDelete. Initially the
+/// acquired location is empty. The 'location' communicated depends on
+/// how the model is being communicated to the agent as indicated by
+/// 'artifact_type'.
+///
+///   TRITONREPOAGENT_ARTIFACT_FILESYSTEM: The location is a directory
+///     on the local filesystem. 'location' returns the full path to
+///     an empty directory that the agent should populate with the
+///     model's artifacts. The returned location string is owned by
+///     Triton, not the agent, and so should not be modified or freed.
+///
+/// \param agent The agent.
+/// \param model The model.
+/// \param artifact_type Returns the artifact type for the location.
+/// \param path Returns the location.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONREPOAGENT_DECLSPEC TRITONSERVER_Error*
+TRITONREPOAGENT_ModelRepositoryLocationAcquire(
+    TRITONREPOAGENT_Agent* agent, TRITONREPOAGENT_AgentModel* model,
+    TRITONREPOAGENT_ArtifactType* artifact_type, const char** location);
```

Comment:
What happens if the function is called multiple time? I assume it will always return the same temporary location?

Reply:
Yes. Also, I'm likely going to change artifact_type to an in/out so that the agent can request a certain type of location (initially there is just "local filesystem") and then triton will prefer to return that type of location (but agent must handle any returned type). With that change will then need to always return same artifact type as well if called multiple times.